pkg/volume/plugins.go

```diff
@@ -121,6 +122,9 @@ type VolumeHost interface {
 	// the provided spec.  See comments on NewWrapperBuilder for more
 	// context.
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
+
+	//Get cloud provider from kubelet
+	GetCloudProvider() cloudprovider.Interface
```

Comment:
What about providers with APIs that aren't cloud providers?   EMC and NetApp are two NFS devices that come to mind that have an API we can use for provisioning.  They're not "cloud providers" and have nothing to do with attaching volumes, but we would need that API client correctly configured and in the plugin for use.